app/services/courses/assign_program_type_service.rb

```diff
@@ -9,6 +9,7 @@ def execute(funding_type, course)
       when "fee"
         course.program_type = update_fee_program(course)
       end
+      course.save unless course.errors.any?
```

Comment:
We don't need the `course.errors.any?` bit. It would be better to just call `course.save` as that would make the return value of the method consistently a boolean that indicates whether it succeeded or not.